### ISSUE TITLE
[BUG]: Latest version 3.1.41 throws an error and crashes

### ISSUE THREAD

**Issue by carlosparreno** [IBMa/equal-access#1149]
### Project

a11y checker engine

### Browser

Chrome, Safari, Firefox

### Operating system

MacOS, Other

### Description

- We use the accessibility checker in a React project. 
- We are using version "^3.1.40" that automatically picks up any new minor/patch version.
- Since the time version 3.1.41 has been released our accessibility tests fail with the following error:
`Error: Cannot find module '@babel/runtime-corejs3/helpers/interopRequireDefault' from '../../../../../tmp/accessibility-checker/engine/ace-node.js'`

- When we fixate the version to be 3.1.40 the error is not reproducible any longer.

- Additionally I add here how we use `accessibility-checker`. All components are used with the following function:

```
  /**
   * Runs the accessibility scan and on failure, outputs the results.
   */

async function runAAT(component, label) {
    const validateComponent = async () => {
      const results = await AAT.getCompliance(component, label);
      if (AAT.assertCompliance(results.report) === 0) {
        return true;
      }
      return false;
    };
    const a11yValidation = await validateComponent();
    if (a11yValidation === true) {
      fs.unlinkSync(`${process.env.PWD}/results/a11y/${label}.json`);
    }
    return a11yValidation;
  }

```

### Steps to reproduce

1. Install "accessiblity-checker": 3.1.41.
2. Have tests using the `getCompliance` and `assertCompliance` functions.
3. Error is thrown:

<img width="1101" alt="Screenshot 2022-11-09 at 16 26 29" src="https://user-images.githubusercontent.com/25571131/200885525-cd673b01-a4ed-4876-949e-048567161359.png">

And after this initial error.... every test fails with the following error message:

<img width="620" alt="Screenshot 2022-11-09 at 16 26 45" src="https://user-images.githubusercontent.com/25571131/200885616-b6ad3c46-8f9c-42bf-b3e6-be835c3cbc26.png">

**Comment by tombrunet** [comment]
Quick test on our end had no failure. We don't have any imports in the ace-node.js file - you can open https://unpkg.com/accessibility-checker-engine@3.1.41/ace-node.js and search and see that that @babel is not referenced anywhere. I'll try to run through some of our boilerplates to see if there's some react combination that's triggering that, but it seems like something in your environment is injecting into that file. Can you possibly include a copy of your ace-node.js file so we can compare against the original?

**Comment by carlosparreno** [comment]
Hi @tombrunet,

I'm using `accessibility-checker` package, not the mentioned `accessibility-checker-engine`, not sure if we are talking about different packages that you guys provide.

I'm not able to find the ace-node.js file in the `node_modules/accessibility-checker`. See following image:

<img width="1171" alt="Screenshot 2022-11-09 at 17 41 09" src="https://user-images.githubusercontent.com/25571131/200901844-cf3fb7fe-05d1-45d1-b430-1257a42fbb3a.png">


The error reported seems to be in a `tmp` folder created by ACEngineManager.js as per my understanding.

Am I making sense?

**Comment by tombrunet** [comment]
`accessibility-checker` fetches the `accessibility-checker-engine` and writes it to that tmp folder so that it can load the engine and run it. If you look at the contents of that file in the /tmp folder, it should match that from unpkg.

**Comment by aliunwala** [comment]
From Triage meeting:
We are still not seeing the babel dependency in the ace-node.js file. (And we know our version of ace-node.js does not have any dependencies) So we think this has to be something you are loading into your environment.

The next thing for you to try is probably loading a minimal set of things and see if you still see this?

**Comment by tombrunet** [comment]
You can add `cacheFolder` to the checker configuration file. 
See https://www.npmjs.com/package/accessibility-checker#Configuration for how to set configuration.

**Comment by elycheea** [comment]
We’ve been having a similar issue in our monorepo. We were able to resolve tests locally after setting the `cacheFolder` but are still seeing failures in our Actions.

https://github.com/carbon-design-system/ibm-cloud-cognitive/actions/runs/3641804091/jobs/6148179111#step:6:23

**Comment by tombrunet** [comment]
@elycheea What did you set as a cacheFolder? Can you try an absolute path?

**Comment by tombrunet** [comment]
That PR should resolve relative paths as absolute, which should be released next week, but that's something you can try if you're in a hurry.

**Comment by elycheea** [comment]
Thanks @tombrunet We’ll give it a shot — not sure if our situation ends up a bit more complicated with our monorepo though. I think my team has reached out before with other questions related to config due to our setup.

**Comment by grabowsd** [comment]
Hi @tombrunet , we’re having issues with setting up the cacheFolder. We tried adding the cacheFolder as follows: 
<img width="709" alt="Screenshot 2022-12-15 at 11 43 32" src="https://user-images.githubusercontent.com/58401975/207867561-cff90ed1-0987-4345-86aa-6dfc6893cad3.png">

but we get the following error for all of our tests:
<img width="1221" alt="Screenshot 2022-12-15 at 11 43 47" src="https://user-images.githubusercontent.com/58401975/207867625-1a92b3ca-babd-4d66-833a-633c9e2f0ce8.png">

 We can see that ace-node.js exists in the path specified in cacheFolder option:
<img width="985" alt="Screenshot 2022-12-15 at 11 40 54" src="https://user-images.githubusercontent.com/58401975/207867762-bb1b0230-5638-43b7-9195-632d29963226.png">

**Comment by tombrunet** [comment]
Can you please try version [3.1.42-rc.0](https://www.npmjs.com/package/accessibility-checker/v/3.1.42-rc.0) using the cacheFolder?

**Comment by grabowsd** [comment]
Thanks @tombrunet.  I tried [3.1.42-rc.0](https://www.npmjs.com/package/accessibility-checker/v/3.1.42-rc.0) but I’m still getting the same error.
<img width="1221" alt="Screenshot 2022-12-15 at 11 43 47" src="https://user-images.githubusercontent.com/58401975/208120102-e75c4b8b-75ab-4710-87ea-8fd9b9e7d062.png">

I noticed that if I change the cacheFolder to just accessibility-checker the tests are passing locally but they fail in Github Actions. This also happens with version 3.1.42-rc.0. 

<img width="667" alt="Screenshot 2022-12-16 at 12 26 36" src="https://user-images.githubusercontent.com/58401975/208120256-570edda9-4bea-4a1b-8ffd-e27c862c0ddb.png">

<img width="1227" alt="Screenshot 2022-12-16 at 14 30 33" src="https://user-images.githubusercontent.com/58401975/208120628-a62b9d3c-7d62-44ed-884b-4512109c2420.png">

**Comment by davidmenendez** [comment]
i'm taking over this issue on my team from @elycheea and wanted to follow up. has there been any update on this?

**Comment by sct-will** [comment]
Fix is in PR: https://github.com/IBMa/equal-access/pull/1240

**Comment by grabowsd** [comment]
Hi at @tombrunet  and @sct-will, 
Thanks for providing a fix.
 
I installed the latest version of accessibility-checker 3.1.43 that I believe contains the fix but I’m getting the following error:    
<img width="1428" alt="Screenshot 2023-02-01 at 15 07 28" src="https://user-images.githubusercontent.com/58401975/216367955-82325ad5-5f80-4b21-8cff-dcbdd31dd194.png">

I’m not sure if this is related to the same issue of cacheFolder in config but it looks like when I’m importing modules from accessibility-checker to run with Jest it cannot find `string-hash` module required in `multiScanData.js`. Is this something that I need to install separately?   

<img width="937" alt="Screenshot 2023-02-02 at 15 26 12" src="https://user-images.githubusercontent.com/58401975/216368642-b5bbdb35-ad26-4f42-ba55-9f466942ce31.png">

**Comment by aliunwala** [comment]
@grabowsd Could you try doing another npm install in the folder you are working in (Probably inside the accessibility-checker folder) 

That should install string-hash (It is included in the package.json here: https://github.com/IBMa/equal-access/blob/master/accessibility-checker/package.json)

**Comment by davidicus** [comment]
> @grabowsd Could you try doing another `npm install` in the folder you are working in (Probably inside the accessibility-checker folder)
> 
> That should install string-hash (It is included in the package.json here: https://github.com/IBMa/equal-access/blob/master/accessibility-checker/package.json)
> 
> If you get any errors after that please post a screen shot again.

@grabowsd This packages is missing from the latest release of `accessibility-checker` as seen [here](https://unpkg.com/browse/accessibility-checker@3.1.43/package.json)

**Comment by tombrunet** [comment]
@aliunwala Those dependencies need to be in this package.json: https://github.com/IBMa/equal-access/blob/master/accessibility-checker/src/package.json. I'll fix.

**Comment by ErickRenteria** [comment]
Closing issue

**Comment by tombrunet** [comment]
@carlosparreno While digging into something else, I found perhaps a cleaner fix for this. Add this to your jest config instead of changing the cache folder
`"transformIgnorePatterns": ["ace-node\\.js"]`
By default, Jest injects babel into all js/jsx files. This tells Jest not to do that for the engine.

**Comment by grabowsd** [comment]
> @carlosparreno While digging into something else, I found perhaps a cleaner fix for this. Add this to your jest config instead of changing the cache folder `"transformIgnorePatterns": ["ace-node\\.js"]` By default, Jest injects babel into all js/jsx files. This tells Jest not to do that for the engine.

@tombrunet, thanks for your help! Adding `  transformIgnorePatterns: ["/node_modules/", "\\.pnp\\.[^\\\/]+$", "ace-node\\.js"],` to our jest config solved the issue. 

Note `"/node_modules/", "\\.pnp\\.[^\\\/]+$"` is `transformIgnorePatterns` default value without which `accessibility-checker` was throwing errors.
